opendevin/server/session/agent.py

```diff
@@ -111,7 +112,7 @@ async def _create_controller(self, start_event: dict):
         )
         try:
             agent_state = State.restore_from_session(self.sid)
-            self.controller._set_initial_state(agent_state)
+            self.controller.set_state(agent_state)
```

Comment:
This might be a merge issue? _set_initial_state() is a new method, formerly set_state, but in agent_controller I see this PR adds the old set_state()

Reply:
set_initial_state combines a few cases where we needed to initialize state in one place, that's all - though this spot here shows it's probably named better without underscore 😅

Reply:
should I combine these into one method and name it without the '_' prefix? the behavior of directly calling the methods that start with '_' is similar to accessing a protected member '_set_initial_state' of a class. although python doesn't explicitly forbid these behaviors, it's not considered good practice. 😄

Reply:
I mean, I think they were already combined. `set_state` was deleted, wasn't it, in `main`?

Yes, we could name it without underscore, precisely. It was initially intended to be "internal" to the controller, but it turns out it's not. 😅